### PR TITLE
CSP-497-Implemented outer API for getProfiles

### PR DESCRIPTION
**Pull request by YuvrajPatil-DFE**
Implemented outer API for getprofiles and respected testcases.

### CODE REVIEW

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/ProfilesController.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        public ProfilesController(IMediator mediator) => _mediator = mediator;
+
+        /// <summary>
+        /// Get list of profiles by user type
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("{userType}")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(GetProfilesByUserTypeQueryResult), StatusCodes.Status200OK)]
```
Comment (MarkFCain) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/ProfilesController.cs:22]: you can add
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]

**Review by MarkFCain — DISMISSED** [review]

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/ProfilesController.cs (hunk 1 above)
Comment (DevenKShah) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/ProfilesController.cs:22]: This is not required.

**Review by MarkFCain — APPROVED** [review]

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/ProfilesControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            result.Should().NotBeNull();
+
+            var model = result?.Value;
+
+            model.Should().BeEquivalentTo(response);
+        }
+
+        [Test]
+        [MoqAutoData]
+        public async Task And_NoProfilesReturnedFromMediator_Then_Return500(
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/ProfilesControllerTests.cs:35]: I am not sure how this test is even working, but it is redundant and must be removed.

**Review by DevenKShah — DISMISSED** [review]

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/ProfilesControllerTests.cs (hunk 2 above)
Comment (YuvrajPatil-DFE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/ProfilesControllerTests.cs:35]: Deleted

**Review by DevenKShah — APPROVED** [review]

**Review by ravihejmadi — APPROVED** [review]

**Pull request merged**